apps/src/templates/HideToolbarHelper.jsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import msg from '@cdo/locale';
+import cookies from 'js-cookie';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+import trackEvent from '../util/trackEvent';
+import _ from 'lodash';
+
+const styles = {
+  closeX: {
+    position: 'absolute',
+    top: 5,
+    right: 5,
+    fontSize: 13
+  }
+};
+
+// Note that additional styling can be found in apps/style/HideToolbarHelper.scss.
+
+const HidetoolbarHelperCookieName = 'hide_toolbar_helper';
+
+/**
+ * An overlay with instrutions on hiding the toolbar for iPhone with iOS 13.
+ */
+export default class HideToolbarHelper extends React.Component {
+  state = {
+    showHelper: false
+  };
+
+  constructor(props) {
+    super(props);
+
+    // Track whether we knew the toolbar was shown last update, so that
+    // if it disappears, we can track the event in analytics...
+    this.wasToolbarShowing = false;
+
+    // ...but only track it once.
+    this.trackedToolbarHide = false;
+  }
+
+  updateLayout = () => {
+    const isiOS13 = navigator.userAgent.indexOf('iPhone OS 13') !== -1;
+    const isHideCookieSet = cookies.get(HidetoolbarHelperCookieName);
+    const isLandscape = window.orientation !== 0;
+
+    // window.innerHeight is smaller than document.body.offsetHeight when
+    // the iOS 13 Safari toolbar is showing.  It becomes equal when the toolbar
+    // is hidden.  (Interestingly, document.documentElement.clientHeight is
+    // the same as document.body.offsetHeight in both cases.)
+    const isToolbarShowing = window.innerHeight < document.body.offsetHeight;
+
+    // Let's track the disapearance of the toolbar, just once.
+    if (
+      this.wasToolbarShowing &&
+      !isToolbarShowing &&
+      !this.trackedToolbarHide
+    ) {
+      trackEvent('Research', 'HideToolbarHelper', 'hid-' + window.innerHeight);
```

Comment:
Clarification question: Why record the `window.innerHeight`?

Reply:
I want to see how often we see a success, and I also want to know the phone heights out in the wild, so I just combined the two.

Reply:
Ok makes sense :) - In that case, do you mind specifying what the numerical value is in the logging message? i.e. I can imagine someone seeing this event get logged without knowing what the number means after 'hid-'

Reply:
(same for the click event below)

Reply:
I'm actually fine leaving those terse.  It's under the `"Research"` category which I previously created for the `"VideoSiteAccessibility"` action, and the events under there also had very terse identifiers.  They're designed to be easier to analyse in bulk, rather than be too human-readable.  If someone's really keen on digging in, they can find a description in this PR :)